src/main/scala/swiss/dasch/domain/MovingImageService.scala

```diff
@@ -42,50 +42,63 @@ case class MovingImageService(storage: StorageService, executor: CommandExecutor
       _       <- executor.executeOrFail(cmd)
     } yield ()
 
-  def extractMetadata(file: MovingImageDerivativeFile, assetRef: AssetRef): Task[MovingImageMetadata] =
+  def extractMetadata(
+    original: Original,
+    derivative: MovingImageDerivativeFile,
+    assetRef: AssetRef
+  ): Task[MovingImageMetadata] =
     for {
-      _       <- ZIO.logInfo(s"Extracting metadata for $file, $assetRef")
-      absPath <- file.toPath.toAbsolutePath
-      cmd <-
-        executor.buildCommand(
-          "ffprobe",
-          s"-v error -select_streams v:0 -show_entries stream=width,height,duration,r_frame_rate -print_format json -i $absPath"
-        )
-      metadata <- executor.executeOrFail(cmd).flatMap(out => parseMetadata(out.stdout))
-    } yield metadata
+      _                          <- ZIO.logInfo(s"Extracting metadata for $derivative, $assetRef")
+      ffprobeInfo                <- extractWithFfprobe(derivative)
+      (dimensions, duration, fps) = ffprobeInfo
+      derivativeMimeType         <- mimeTypeGuesser.guess(derivative.toPath)
+      originalMimeType           <- mimeTypeGuesser.guess(Path(original.originalFilename.value))
```

Comment:
if the function returned the option directly, this could just be with `=` instead of `<-`

Reply:
See abf318958def7a285af4413dc32d24383c304771